### ISSUE TITLE
`jsx-no-leaked-render` performs redundant props check and invalid fix in coerce mode

### ISSUE THREAD

**Issue by wialy** [jsx-eslint/eslint-plugin-react#3431]
# Context

There is a sample `Checkbox` component:

```typescript
const Checkbox = ({
  isChecked = false,
  isIndeterminate = false,
}: {
  isChecked?: boolean;
  isIndeterminate?: boolean;
}) => (
  <input 
    checked={isIndeterminate ? false : isChecked} 
    type="checkbox"
  />
);
```

The eslint config file contains a rule:

```
"react/jsx-no-leaked-render": ["error", { validStrategies: ["coerce"] }],
```

# Expected behavior

Linting the code raises no issues

# Actual behavior 

1. Eslint displays an error for the line:

```typescript
checked={isIndeterminate ? false : isChecked}
```

```
Potential leaked value that might cause unintentionally rendered values or rendering crasheseslint[react/jsx-no-leaked-render](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/jsx-no-leaked-render.md)
```

2. When auto-fix is performed, the code above is replaced with

```typescript
checked={!!isIndeterminate && false} 
```

which causes the logical issues (the `isChecked` is removed in that case).

**Comment by ljharb** [comment]
isChecked could be undefined, according to the types, and default arguments aren’t something that can really be used reliably with static analysis - so i think the warning may be correct.

I agree, though, that the autofix is incorrect - it should be fixing it to `!isIndeterminate && isChecked`

**Comment by ljharb** [comment]
cc @Belco90

**Comment by himanshu007-creator** [comment]
hey, would like to work on this issue

**Comment by ljharb** [comment]
Go for it!

**Comment by himanshu007-creator** [comment]
i was not able to reproduce the issue

**Comment by ljharb** [comment]
In that case, can you make a PR with passing test cases, and it can close this issue?

**Comment by tcl333** [comment]
I had an identical situation to the OP, and this error still occurred. @himanshu007-creator, did you remember to change the allowed strategies to only `coerce`?
